src/Hypsography/Hypsography.jl

```diff
@@ -64,180 +122,156 @@ If the decay-scale is poorly specified (i.e., `s * zₜ` is lower than the maxim
-        face_column = column(face_local_geometry, i, j, h)
-        fZ_column = column(Fields.field_values(fZ), i, j, h)
-        f∇Z_column = column(Fields.field_values(f∇Z), i, j, h)
-        center_column = column(center_local_geometry, i, j, h)
-        cZ_column = column(Fields.field_values(cZ), i, j, h)
-        c∇Z_column = column(Fields.field_values(c∇Z), i, j, h)
-
-        # update face metrics
-        for v in 1:(Nv + 1)
-            local_geom = face_column[v]
-            coord = if coord_type <: Geometry.Abstract2DPoint
-                c1 = Geometry.components(local_geom.coordinates)[1]
-                coord_type(c1, fZ_column[v])
-            elseif coord_type <: Geometry.Abstract3DPoint
-                c1 = Geometry.components(local_geom.coordinates)[1]
-                c2 = Geometry.components(local_geom.coordinates)[2]
-                coord_type(c1, c2, fZ_column[v])
-            end
-            Δz = if v == 1
-                # if this is the domain min face level compute the metric
-                # extrapolating from the bottom face level of the domain
-                2 * (cZ_column[v] - fZ_column[v])
-            elseif v == Nv + 1
-                # if this is the domain max face level compute the metric
-                # extrapolating from the top face level of the domain
-                2 * (fZ_column[v] - cZ_column[v - 1])
-            else
-                cZ_column[v] - cZ_column[v - 1]
-            end
-            ∂x∂ξ = reconstruct_metric(local_geom.∂x∂ξ, f∇Z_column[v], Δz)
-            W = local_geom.WJ / local_geom.J
-            J = det(Geometry.components(∂x∂ξ))
-            face_column[v] = Geometry.LocalGeometry(coord, J, W * J, ∂x∂ξ)
-        end
 
-        # update center metrics
-        for v in 1:Nv
-            local_geom = center_column[v]
-            coord = if coord_type <: Geometry.Abstract2DPoint
-                c1 = Geometry.components(local_geom.coordinates)[1]
-                coord_type(c1, cZ_column[v])
-            elseif coord_type <: Geometry.Abstract3DPoint
-                c1 = Geometry.components(local_geom.coordinates)[1]
-                c2 = Geometry.components(local_geom.coordinates)[2]
-                coord_type(c1, c2, cZ_column[v])
-            end
-            Δz = fZ_column[v + 1] - fZ_column[v]
-            ∂x∂ξ = reconstruct_metric(local_geom.∂x∂ξ, c∇Z_column[v], Δz)
-            W = local_geom.WJ / local_geom.J
-            J = det(Geometry.components(∂x∂ξ))
-            center_column[v] = Geometry.LocalGeometry(coord, J, W * J, ∂x∂ξ)
-        end
-    end
+    center_∇Z_field =
+        grad.(
+            Fields.Field(
+                center_z_local_geometry,
+                center_flat_space,
+            ).coordinates.z
+        )
+    Spaces.weighted_dss!(center_∇Z_field)
+
+    face_∇Z_field =
+        grad.(
+            Fields.Field(face_z_local_geometry, face_flat_space).coordinates.z
+        )
+    Spaces.weighted_dss!(face_∇Z_field)
```

Comment:
```suggestion
    face_∇Z_field =
        grad.(
            Fields.Field(face_z_local_geometry, face_flat_space).coordinates.z
        )
    buffer = (;
        c = Spaces.create_dss_buffer(center_∇Z_field),
        f = Spaces.create_dss_buffer(face_∇Z_field)
    )

    Spaces.weighted_dss!(
        center_∇Z_field => buffer.c,
        face_∇Z_field => buffer.f
    )
```
I don't know if there is a buffer-less version, but this is a better way to call dss.